core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java

```diff
@@ -136,10 +138,15 @@ public static void run(Application application, Class<? extends QuarkusApplicati
                 }
             }
         } catch (Exception e) {
-            if (appStarted) {
-                //we only log if the error occurred after the application was started
-                //as the generated application class already has logging
-                Logger.getLogger(Application.class).error("Error running Quarkus application", e);
+            if (exitCodeHandler == null) {
+                if (appStarted) {
+                    //we only log if the error occurred after the application was started
+                    //as the generated application class already has logging
+                    Logger.getLogger(Application.class).error("Error running Quarkus application", e);
+                } else {
+                    Logger.getLogger(Application.class).errorv(e, "Failed to start application (with profile {0})",
```

Comment:
I don't think the if statement is necessary here any more.

Reply:
I remember having `appStarted` == `true` indicating that the application was started but failed for some reason afterward. But I could not make it happen again. I thought the if the statement was intentional to differentiate errors in the different stages.

Going to remove.